File: src/main/kotlin/a/Magnets.kt
package a

/* Problem Link: https://codeforces.com/problemset/problem/344/A */

fun main() {
    val n = readln().toInt()
    val magnets = mutableListOf<String>()
    var groups = 1

    for (i in 0..<n) {
        val x = readln()
        magnets.add(x)
        if (i > 0 && magnets[i] != magnets[i - 1]) groups += 1
    }

    print(groups)
}

File: Codeforces/C++/problemset/Magnets.cpp
// Problem Link: https://codeforces.com/problemset/problem/344/A

#include <iostream>
using namespace std;
typedef long long ll;
int main()
{
    ll n, g = 1;
    cin >> n;
    ll *m = new ll[n];
    for(ll i = 0; i < n; i++)
    {
        cin >> m[i];
        if(i > 0 && m[i] != m[i-1]) g++;
    }
    cout << g;
    delete []m;
    return 0;
}
